### DISCUSSION TITLE
Mobile Team Project Plan

### DISCUSSION THREAD

**Discussion by cahidenes** [bounswe/bounswe2023group2#283] [General]
Discussion for mobile application.

**Comment by HasanBingolbali** [comment]
We are decided to use MMVM. We are going to need manager classes to manage data. I propose writing NetworkManager and FileManager classes for reusable and data control. We also need to setup navigation system to avoid confusion. Also for UI we may need to define some standarts for consistency

**Comment by cahidenes** [comment]
First draft of plan:

## Empty Pages
- [x] Create Empty Application
- [x] Create Navigation Structure
- [x] Fill Login page, Signup page, Profile page and Add Activity page
## Establishing Local Database
- [ ] Establish database
- [ ] Create tables
## Establishing Network
- [x] Implementing network manager
- [ ] Create a structure to communicate endpoints
## Implementing login page
- [ ] Implement UI
- [ ] Implement communication with API

**Reply by cahidenes** [reply]
Other things that should be done:
- Other pages
- Notification
- Map
- Testing

We should list all things and share between milestone 2 and final milestone until Thursday.

**Comment by egecans** [comment]
We can use a navigation component or just add a Fragment to the Activity. If we don't need multiple activities for some purpose, we should use a single activity design pattern. Multiple fragments could be appended into a single activity on Android.

**Comment by egecans** [comment]
You can watch this playlist for beginners. I strongly recommend [that](https://www.youtube.com/playlist?list=PLQkwcJG4YTCTq1raTb5iMuxnEB06J1VHX)

**Reply by higurbuzz** [reply]
Thanks for sharing.
